src/mavedb/lib/identifiers.py

```diff
@@ -149,7 +149,7 @@ def reference_html(self) -> str:
         else:
             doi_str = "" if not self.preprint_doi else self.preprint_doi
             title = "(None)" if not self.title else self.title.strip(".")
-            journal = "(None)" if not self.publication_journal else self.publication_journal.strip(".")
+            journal = "(None)" if not (hasattr(self, "publication_journal") and self.publication_journal) else self.publication_journal.strip(".")
```

Comment:
I must admit this change mystifies me a bit. It reads identically to the previous code for me. What am I missing?

Reply:
the "publication_journal" attribute doesn't always exist, which means checking "not self.publication_journal" throws an AttributeError.  `hasattr(self, "publication_journal")` checks that the attribute exists ... it might still be None or otherwise false but at least we know it exists before we try to check its value.

This is ... an unusual way to do things.
I think it would be better to initialize it (and other attributes) to None if it isn't appropriate, but I don't know if that would affect anything else in the code so I just did it this way as a workaround.
@bencap might have some input here?

Reply:
I didn't initialize those fields to make it clear in the `ExternalPublication` object that those fields were not defined for a particular publication on some external source of data, rather than just empty. The `ExternalPublication` model only ever acts as a temporary container when converting between some returned information from the PubMed / ___rxiv APIs. I had believed it would be more useful to raise an error when trying to access those fields and that `reference_html` issue got past me.

If we were to initialize those attributes to `None` instead, it shouldn't effect any other parts of the codebase. Places that would try to access a non-existent attribute are getting some sort of blanket default at the moment (https://github.com/VariantEffect/mavedb-api/blob/149c101457cb179062caf728382441852f6edbf1/src/mavedb/lib/identifiers.py#L304C19-L304C19).